src/algorithms/telemetry_decoder/gnuradio_blocks/sbas_l1_telemetry_decoder_gs.cc

```diff
@@ -96,7 +120,33 @@ void sbas_l1_telemetry_decoder_gs::set_satellite(const Gnss_Satellite &satellite
 void sbas_l1_telemetry_decoder_gs::set_channel(int32_t channel)
 {
     d_channel = channel;
-    LOG(INFO) << "SBAS channel set to " << channel;
+    DLOG(INFO) << "Navigation channel set to " << channel;
+    // ############# ENABLE DATA FILE LOG #################
```

Comment:
You've created an interesting dependency here. Now, this function is responsible for opening (truncating) the dump file. If this function is never called, will the dump file be created elsewhere?

If the dump file was previously opened and then this function is called, the internal state describing the channel will be updated but the dump-file will continue to be the old channel. That can't be good. It seems that setting the channel and opening (re-opening?) the dump file are separate concerns. Perhaps they shouldn't be mixed here.